pkg/helm/releases.go

```diff
@@ -37,3 +43,70 @@ func MakeHelmRelease(name, version, cluster, namespace string, helmRepository ty
 		},
 	}
 }
+
+// FindReleaseInNamespace iterates through a slice of HelmReleases to find one with a given name in a given namespace, and returns it with its index.
+func FindReleaseInNamespace(existingReleases []helmv2beta1.HelmRelease, name, ns string) (*helmv2beta1.HelmRelease, int, error) {
+	for i, r := range existingReleases {
+		if r.Name == name && r.Namespace == ns {
+			return &r, i, nil
+		}
+	}
+
+	return nil, -1, nil
+}
+
+// AppendHelmReleaseToString appends a HelmRelease to a string.
+func AppendHelmReleaseToString(content string, newRelease *helmv2beta1.HelmRelease) (string, error) {
+	var sb strings.Builder
+	if content != "" {
+		sb.WriteString(content + "\n")
+	}
+
+	helmReleaseManifest, err := kyaml.Marshal(newRelease)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal HelmRelease: %w", err)
+	}
+
+	sb.WriteString("---\n" + string(helmReleaseManifest))
+
+	return sb.String(), nil
+}
+
+// SplitHelmReleaseYAML splits a manifest file that contains one or more Helm Releases that may be separated by '---'.
+func SplitHelmReleaseYAML(resources []byte) ([]helmv2beta1.HelmRelease, error) {
+	var helmReleaseList []helmv2beta1.HelmRelease
+
+	decoder := apimachinery.NewYAMLOrJSONDecoder(bytes.NewReader(resources), 100000000)
+
+	for {
+		var value helmv2beta1.HelmRelease
+		if err := decoder.Decode(&value); err != nil {
+			if err == io.EOF {
+				break
+			}
+
+			return nil, err
+		}
+
+		helmReleaseList = append(helmReleaseList, value)
+	}
+
+	return helmReleaseList, nil
+}
+
+func PatchHelmRelease(existingReleases []helmv2beta1.HelmRelease, patchedHelmRelease helmv2beta1.HelmRelease, index int) (string, error) {
+	existingReleases[index] = patchedHelmRelease
```

Comment:
We could switch to passing around `[]*helmv2beta1.HelmRelease` instead, and then just edit the list in-place. Then when it comes to patching the list has already been modified, so this func just becomes marshaling the list of helm releases into the string

Reply:
done! :)